src/components/Login.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useContext } from "react";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import { createPatient } from "../firebase";
+import { jsPsych } from "jspsych-react";
+import "../App.css";
+import "bootstrap/dist/css/bootstrap.css";
+import "@fortawesome/fontawesome-free/css/all.css";
+import LoginContext from "../contexts/LoginContext";
+
+function Login() {
+  // State variables for login screen
+  const dateTimestamp = Date.now();
+  const curDate = new Date(dateTimestamp);
+  const [patientId, setPatient] = useState("");
+  const [studyId, setStudy] = useState("");
+  const [startDate] = useState(curDate.toString());
+  const [timestamp] = useState(dateTimestamp.toString());
+
+  // Take login context
+  const { login } = useContext(LoginContext);
+
+  // Checks if forms are filled in
+  function validateForm() {
+    return patientId.length > 0 && studyId.length > 0;
+  }
+
+  // Function to log in patient
+  function handleSubmit(e) {
+    e.preventDefault();
+    const logPatient = async () => {
+      // Checks if patient exists in firestore
+      const loggedIn = await createPatient(
+        patientId,
+        studyId,
+        startDate,
+        timestamp
+      );
+
+      // If patient and study are registered
+      if (loggedIn) {
+        jsPsych.data.addProperties({
+          patient_id: patientId,
+          study_id: studyId,
+          timestamp: timestamp,
+          start_date: startDate,
+        });
+        login();
+      }
+    };
+    logPatient();
```

Comment:
would it be cleaner to do this with callbacks than with the nested async function? @broarr @mirestrepo

Reply:
I don't have strong feelings about the async/await here, but I'm not an expert on async.... I'm wondering though if `logPatient` could live in the `firebase.json` file were `createPatient` is?

Reply:
Also I think we decided to use Participant instead of Patient. Right?

Reply:
And one last question is what happens if the login screen is shown when `ONLINE=false` and the participant ID is not passed as an `ENV`? The code here for `logPatient` is firebase specific, so do we need to make sure we check for that?

Reply:
good point, there should be a check for FIREBASE

Reply:
also, need the logic to get the mturk id still I think (unless I missed that)

Reply:
I think we need the jspsych updates here (as that's global to all use cases), but need more ways to set loggedIn (and think about how to make that cleanest/most re-usable/abstract

Reply:
@mirestrepo @mcmcgrath13 How would checking for FIREBASE look like? If FIREBASE == false, do we just let them enter any credentials in the login screen to continue? 
The logic for MTURK is in the useEffect in app and I just added setLogin(true) when MTURK. 
What I had in mind is for the logic for other login to be in useEffect in App.js and login.js to be the login screen, which is pretty much tied to FIREBASE.

Reply:
In my mind the login for Firebase does a credentials check, the login for Electron just sets variables - unless they were set by the ENVS..

Reply:
Kind of the `if FIREBASE` is just around the `createPatient` call or something like that?